### ISSUE TITLE
[Windows] Server-side debugger doesn't stop in breakpoints after v15.0.3-canary.7

### ISSUE THREAD

**Issue by emilioestebanez** [vercel/next.js#73214]
### Link to the code that reproduces this issue

https://github.com/emilioestebanez/next-15-issues

### To Reproduce

1. Open in VSCode
2. Go to `Run and Debug` and execute the `Next.js: debug server-side`
3. Set a breakpoint in `src/app/page.tsx`
4. Visit http://localhost:3000

You'll see that the debugger doesn't stop in the breakpoint in Windows.

Now, if you go to `package.json` and remove `--turbopack` from the `dev` script and try the steps above again, you'll see that the debugger is able to stop in the breakpoint correctly.

This happens after next@15.0.3-canary.7

This has been tested also in 15.2.2 and it still doesn't work.

### Current vs. Expected behavior

I expect the breakpoint to stop with and without `--turbopack` enabled.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 65208
  Available CPU cores: 20
Binaries:
  Node: 20.18.0
  npm: 10.8.2
  Yarn: 1.22.22
  pnpm: 9.12.3
Relevant Packages:
  next: 15.0.3 // Latest available version is detected (15.0.3).
  eslint-config-next: 15.0.3
  react: 19.0.0-rc-66855b96-20241106
  react-dom: 19.0.0-rc-66855b96-20241106
  typescript: 5.7.2
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

I think this is the cause of the issue:
https://github.com/vercel/next.js/pull/71984

**Comment by michaelschufi** [comment]
I've checked out the repo because it was mentioned in #73258.

It's _working inside WSL 2_ (Linux, see below) but _**not** on Windows_ 10 itself.
I've also tried the canary version, which has the same issue.

**Canary version tested 🔢**
```
"dependencies": {
  "react": "19.0.0-rc-b01722d5-20241114",
  "react-dom": "19.0.0-rc-b01722d5-20241114",
  "next": "15.0.4-canary.30"
}
```

**Not working on ❌**
```
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
  Available memory (MB): 32479
  Available CPU cores: 16
Binaries:
  Node: 22.11.0
  npm: 10.9.0
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.4-canary.30 // Latest available version is detected (15.0.4-canary.30).
  eslint-config-next: 15.0.3
  react: 19.0.0-rc-b01722d5-20241114
  react-dom: 19.0.0-rc-b01722d5-20241114
  typescript: 5.7.2
Next.js Config:
  output: N/A
```

**Working on ✅**
```
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Fri Mar 29 23:14:13 UTC 2024
  Available memory (MB): 15853
  Available CPU cores: 16
```

**Comment by emilioestebanez** [comment]
> I've checked out the repo because it was mentioned in #73258.
> 
> It's _working inside WSL 2_ (Linux, see below) but _**not** on Windows_ 10 itself. I've also tried the canary version, which has the same issue.
> 
> **Canary version tested 🔢**
> 
> ```
> "dependencies": {
>   "react": "19.0.0-rc-b01722d5-20241114",
>   "react-dom": "19.0.0-rc-b01722d5-20241114",
>   "next": "15.0.4-canary.30"
> }
> ```
> 
> **Not working on ❌**
> 
> ```
> Operating System:
>   Platform: win32
>   Arch: x64
>   Version: Windows 10 Pro
>   Available memory (MB): 32479
>   Available CPU cores: 16
> Binaries:
>   Node: 22.11.0
>   npm: 10.9.0
>   Yarn: N/A
>   pnpm: N/A
> Relevant Packages:
>   next: 15.0.4-canary.30 // Latest available version is detected (15.0.4-canary.30).
>   eslint-config-next: 15.0.3
>   react: 19.0.0-rc-b01722d5-20241114
>   react-dom: 19.0.0-rc-b01722d5-20241114
>   typescript: 5.7.2
> Next.js Config:
>   output: N/A
> ```
> 
> **Working on ✅**
> 
> ```
> Operating System:
>   Platform: linux
>   Arch: x64
>   Version: #1 SMP Fri Mar 29 23:14:13 UTC 2024
>   Available memory (MB): 15853
>   Available CPU cores: 16
> ```

Thanks for checking it!

**Comment by emilioestebanez** [comment]
@Falven , for me it fails with both npm and pnpm.

**Comment by jacksierkstra** [comment]
I am also facing the problem of unbound breakpoints in Visual Studio Code. I am using Yarn. 

Relevant next info:
```
Node.js v22.9.0

Operating System:
  Platform: win32
  Arch: arm64
  Version: Windows 11 Pro
  Available memory (MB): 32072
  Available CPU cores: 12
Binaries:
  Node: 22.9.0
  npm: 10.8.3
  Yarn: 4.5.3
  pnpm: N/A
Relevant Packages:
  next: 15.1.2
  eslint-config-next: 15.1.0
  react: 19.0.0
  react-dom: 19.0.0
  typescript: 5.7.2
Next.js Config:
  output: standalone
``` 

Relevant launch configuration in Visual Studio Code:
```
...
{
      "command": "yarn run dev",
      "name": "Debug: Next.JS",
      "request": "launch",
      "sourceMaps": true,
      "type": "node-terminal",
      "serverReadyAction": {
        "action": "debugWithChrome",
        "killOnServerStop": true,
        "pattern": "- Local:.+(https?://.+)",
        "uriFormat": "%s",
        "webRoot": "${workspaceFolder}"
      },
      "resolveSourceMapLocations": [
        "${workspaceFolder}/**",
        "!**/node_modules/**"
      ]
    }
...
```

Whenever I switch back to the normal dev server (without Turbopack), breakpoints are being bound and hit.

**Comment by raajeshmani** [comment]
This is not a windows only issue. Occurs on macOS as well. Although it is rightly pointed out that it happens only after `v15.03-canary.7` with turbopack. Here's my `launch.json` which works with next `15.0.2` mentioned in [discussion#4860](https://github.com/vercel/turborepo/discussions/4860#discussioncomment-11783098)

```
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.6.0: Wed Jul 31 20:49:39 PDT 2024; root:xnu-10063.141.1.700.5~1/RELEASE_ARM64_T6000
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 20.11.1
  npm: 10.2.4
  Yarn: N/A
  pnpm: 9.14.4
Relevant Packages:
  next: 15.0.2 // There is a newer version (15.1.4) available, upgrade recommended!
  eslint-config-next: 14.2.3
  react: 19.0.0
  react-dom: 19.0.0
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

**Comment by nacholupotti** [comment]
If someones is going here, update your next version to 15.2!!

On that version problem is fixed

**Comment by jacksierkstra** [comment]
> If someones is going here, update your next version to 15.2!!
> 
> On that version problem is fixed

I am on Next 15.2.2 and the problem is not fixed.

**Comment by emilioestebanez** [comment]
Same here, in Next.js 15.2.2 the same issue happens.

**Comment by Godrules500** [comment]
I'm having the same issue. VScode won't hit breakpoints when using turbo.

**Comment by ludekvodicka** [comment]
same issue here. any solution for this?

**Comment by timneutkens** [comment]
Can you give next@canary a try? We've applied a fix for sourcemap paths on Windows in Turbopack.

**Comment by emilioestebanez** [comment]
@timneutkens it works correctly on v15.4.0-canary.7 and newer. Thanks a lot!

I guess this is the PR that fixed it:
https://github.com/vercel/next.js/pull/78453

**Comment by timneutkens** [comment]
Indeed that's the one, thanks for checking Emilio, appreciated!

**Comment by bernatfortet** [comment]
I'm still having the issue on mac and `"next": "^15.4.0-canary.34",`

**Comment by timneutkens** [comment]
Please open a new issue with a reproduction an steps of what isn't working @bernatfortet.

**Comment by bernatfortet** [comment]
Done: https://github.com/microsoft/vscode-js-debug/issues/2227. Something tells me there's probably a simple temporary fix if we can point the debugger to the right sourcefiles. Has anyone been able to figure it out?
